### PR TITLE
Refactor _setup_cli_logging code

### PR DESCRIPTION
**Pull request by thisch**
Change the indentation in _setup_cli_logging by moving the
self._log_cli_enabled check outside of the _setup_cli_logging method.

Thanks for submitting a PR, your contribution is really appreciated!

Here's a quick checklist that should be present in PRs (you can delete this text from the final description, this is
just a guideline):

- [ ] Create a new changelog file in the `changelog` folder, with a name like `<ISSUE NUMBER>.<TYPE>.rst`. See [changelog/README.rst](https://github.com/pytest-dev/pytest/blob/master/changelog/README.rst) for details.
- [ ] Target the `master` branch for bug fixes, documentation updates and trivial changes.
- [ ] Target the `features` branch for new features and removals/deprecations.
- [ ] Include documentation when adding new features.
- [ ] Include new tests or update existing tests when applicable.

Unless your change is trivial or a small documentation fix (e.g.,  a typo or reword of a small section) please:

- [ ] Add yourself to `AUTHORS` in alphabetical order;

### CODE REVIEW

File: src/_pytest/logging.py [hunk 1]
```diff
@@ -420,54 +420,53 @@ def __init__(self, config):
...
-            )
-            self.log_cli_handler = log_cli_handler
-            self.live_logs_context = lambda: catching_logs(
-                log_cli_handler, formatter=log_cli_formatter, level=log_cli_level
+        config = self._config
+        terminal_reporter = config.pluginmanager.get_plugin("terminalreporter")
+        capture_manager = config.pluginmanager.get_plugin("capturemanager")
+
+        if terminal_reporter is None or capture_manager is None:
+            # wouldn't it be better if we raise here?
```
Comment (nicoddemus) [inline src/_pytest/logging.py]: Not sure, those plugins might be disabled for other reasons (for example xdist disables terminal writer).

File: src/_pytest/logging.py [hunk 2]
```diff
@@ -420,54 +420,53 @@ def __init__(self, config):
...
+            return
+
+        # FIXME don't set verbosity level and derived attributes of
```
Comment (nicoddemus) [inline src/_pytest/logging.py]: Please add a comment here pointing to #4733

Comment (thisch) [reply src/_pytest/logging.py]: I'll take care of #4733, once this is merged.

Comment (nicoddemus) [reply src/_pytest/logging.py]: Oh great, no need to update the comment right now then. 👍

**Comment by thisch** [comment]
> Looks good, we also need a changelog entry. Seems to me like a "bugfix" about disabling logging automatically in case capture is also disabled, something which did not work previously anyway?

I just had a look again and checked what happens if the capturemanager plugin is disabled. The live logging still works (because the _LiveLoggingStreamHandler supports that self.capture_manager is None.)

So this PR actually does not fix a bug. Therefore, I decided not to write a bugfix changelog entry. 

I'll adapt the code s.t. the current behavior is not changed.

**Comment by nicoddemus** [comment]
> So this PR actually does not fix a bug. Therefore, I decided not to write a bugfix changelog entry.

I agree, thanks for the explanation. 👍

**Pull request merged**